src/passes/OptimizeInstructions.cpp

```diff
@@ -1299,6 +1299,27 @@ struct OptimizeInstructions
         }
       }
     }
+    if (type.isFloat()) {
+      auto value = right->value.getFloat();
+      if (value == 0.0) {
+        if (binary->op == Abstract::getBinary(type, Abstract::Sub)) {
+          if (std::signbit(value)) {
+            // x - (-0.0)   ==>   x + 0.0
```

Comment:
In what situations is it illegal to do `x + 0.0   ==>   x`?

Reply:
when `x` equal `-0.0`. `-0.0 + 0.0 != -0.0` in compare it in bit-by-bit. LLVM/GCC btw also don't  apply this simplification in the same reason.

Reply:
We should add a fast math flags at some point if we don't already have one because some of these edge cases are ridiculous.

Reply:
Yes, totally agree. This mode is definitely necessary